clar2wasm/src/wasm_generator.rs

```diff
@@ -2031,19 +2030,115 @@ impl WasmGenerator {
         Ok(())
     }
 
-    fn visit_call_user_defined(
+    fn traverse_call_user_defined(
```

Comment:
Thanks for that input. I think in that case, I'd prefer to just repeat the call in each case rather than add the closure. I'll play with this code to see if I can rework it to something that might be better to read.